src/signing_key.rs

```diff
@@ -106,6 +115,71 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> TryFrom<PrivateKeyInfo<'a>> for SigningKey {
+    type Error = Error;
+    fn try_from(pki: PrivateKeyInfo) -> Result<Self, Error> {
+        if pki.algorithm == ALGORITHM_ID {
+            SigningKey::try_from(pki.private_key)
+        } else {
+            Err(Error::MalformedSecretKey)
+        }
+    }
+}
+
+impl ToPrivateKey for SigningKey {
+    fn to_pkcs8_der(&self) -> PrivateKeyDocument {
+        // In RFC 8410, the octet string containing the private key is encapsulated by
+        // another octet string. Just add octet string bytes to the key.
+        let octetstring_bytes_string = "0420";
+        let mut octetstring_array = [0u8; 2];
+        hex::decode_to_slice(octetstring_bytes_string, &mut octetstring_array as &mut [u8]).ok();
+
+        let mut final_key = [0 as u8; 34];
+        let mut key_byte_sequence =  octetstring_array.iter()
+            .chain(self.seed.iter());
+        let _bytes_written = key_byte_sequence
+            .by_ref()
+            .zip(final_key.as_mut())
+            .fold(0, | cnt, (item, slot) | {
+                *slot = item.clone(); cnt+1
+            });
+
+        PrivateKeyInfo {
+            algorithm: ALGORITHM_ID,
+            private_key: &final_key,
+        }.into()
```

Comment:
In `pkcs8` v0.7, you'll want to use [`PrivateKeyInfo::new`](https://docs.rs/pkcs8/0.7.0/pkcs8/struct.PrivateKeyInfo.html#method.new) for this.